Paths to direct file not resolved

Hello,

I am updating my Angular project and its dependencies.

`eslint-import-resolver-typescript@3.1.4` is not able to resolve aliases to direct file versions anymore.

`app.module.ts`
```typescript
import { HeaderModule } from '~app/core/header/header.module'; // This work
import { environment } from '~environment'; // This doesn't work anymore.
```

`tsconfig.json`
```json
   "paths": {
      "~environment": ["src/environments/environment"], // Alias to `src/environments/environment.ts` file
      "~shared/*": ["src/app/shared/*" ],
      "~app/*": ["src/app/*" ],
      "~mocks/*": ["src/tests/mocks/*"]
    }
```

Output:
```
...\angular-seed\src\app\app.module.ts
  11:29  error  Unable to resolve path to module '~environment'  import/no-unresolved

✖ 1 problem (1 error, 0 warnings)
```

I rolback my version to `^2.7.0` et ça fonctionne.

--- a/.changeset/strong-beans-allow.md
+++ b/.changeset/strong-beans-allow.md
@@ -1,0 +1,5 @@
+---
+"eslint-import-resolver-typescript": patch
+---
+
+fix: auto try extensions

--- a/.eslintignore
+++ b/.eslintignore
@@ -1,3 +1,4 @@
+dist
 lib
 CHANGELOG.md
 !/.*.cjs

--- a/.github/workflows/pkg-size.yml
+++ b/.github/workflows/pkg-size.yml
@@ -17,7 +17,7 @@ jobs:
       - name: Setup Node.js
         uses: actions/setup-node@v3
         with:
-          node-version: 16.x
+          node-version: 16
           cache: yarn
 
       - name: Package Size Report

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -16,10 +16,10 @@ jobs:
           # This makes Actions fetch all Git history so that Changesets can generate changelogs with the correct commits
           fetch-depth: 0
 
-      - name: Setup Node.js 16.x
+      - name: Setup Node.js 16
         uses: actions/setup-node@v3
         with:
-          node-version: 16.x
+          node-version: 16
           cache: yarn
 
       - name: Install Dependencies

--- a/README.md
+++ b/README.md
@@ -30,6 +30,8 @@ This means you can:
 - [Installation](#installation)
 - [Configuration](#configuration)
 - [Contributing](#contributing)
+- [Sponsors](#sponsors)
+- [Backers](#backers)
 - [Changelog](#changelog)
 - [License](#license)
 
@@ -108,6 +110,18 @@ Add the following to your `.eslintrc` config:
 We have [GitHub Actions](https://github.com/import-js/eslint-import-resolver-typescript/actions) which will run the above commands on your PRs.
 
 If either fails, we won't be able to merge your PR until it's fixed.
+
+## Sponsors
+
+| 1stG                                                                                                                               | RxTS                                                                                                                               | UnTS                                                                                                                               |
+| ---------------------------------------------------------------------------------------------------------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------- |
+| [![1stG Open Collective backers and sponsors](https://opencollective.com/1stG/organizations.svg)](https://opencollective.com/1stG) | [![RxTS Open Collective backers and sponsors](https://opencollective.com/rxts/organizations.svg)](https://opencollective.com/rxts) | [![UnTS Open Collective backers and sponsors](https://opencollective.com/unts/organizations.svg)](https://opencollective.com/unts) |
+
+## Backers
+
+| 1stG                                                                                                                             | RxTS                                                                                                                             | UnTS                                                                                                                             |
+| -------------------------------------------------------------------------------------------------------------------------------- | -------------------------------------------------------------------------------------------------------------------------------- | -------------------------------------------------------------------------------------------------------------------------------- |
+| [![1stG Open Collective backers and sponsors](https://opencollective.com/1stG/individuals.svg)](https://opencollective.com/1stG) | [![RxTS Open Collective backers and sponsors](https://opencollective.com/rxts/individuals.svg)](https://opencollective.com/rxts) | [![UnTS Open Collective backers and sponsors](https://opencollective.com/unts/individuals.svg)](https://opencollective.com/unts) |
 
 ## Changelog
 

--- a/package.json
+++ b/package.json
@@ -8,6 +8,38 @@
   "contributors": [
     "JounQin (https://www.1stG.me) <admin@1stg.me>"
   ],
+  "donate": {
+    "recipients": [
+      {
+        "name": "unts",
+        "platform": "opencollective",
+        "address": "https://opencollective.com/unts",
+        "weight": 60
+      },
+      {
+        "name": "rxts",
+        "platform": "opencollective",
+        "address": "https://opencollective.com/rxts",
+        "weight": 20
+      },
+      {
+        "name": "1stG",
+        "email": "i@1stg.me",
+        "weight": 20,
+        "platforms": [
+          {
+            "platform": "opencollective",
+            "address": "https://opencollective.com/1stG"
+          },
+          {
+            "platform": "patreon",
+            "address": "https://www.patreon.com/1stG"
+          }
+        ]
+      }
+    ]
+  },
+  "funding": "https://opencollective.com/unts",
   "license": "ISC",
   "packageManager": "yarn@1.22.19",
   "engines": {
@@ -71,7 +103,7 @@
     "synckit": "^0.7.1"
   },
   "devDependencies": {
-    "@1stg/lib-config": "^7.2.0",
+    "@1stg/lib-config": "^7.2.3",
     "@changesets/changelog-github": "^0.4.5",
     "@changesets/cli": "^2.23.0",
     "@mozilla/glean": "^1.0.0",

--- a/src/index.ts
+++ b/src/index.ts
@@ -252,6 +252,7 @@ const isFile = (path?: string | undefined): path is string => {
  * @param {string} file the importing file's full path; i.e. '/usr/local/bin/file.js'
  * @returns The mapped path of the module or undefined
  */
+// eslint-disable-next-line sonarjs/cognitive-complexity
 function getMappedPath(
   source: string,
   file: string,
@@ -265,7 +266,16 @@ function getMappedPath(
       paths = [resolved]
     }
   } else {
-    paths = mappers!.flatMap(mapper => mapper?.(source)).filter(isFile)
+    paths = mappers!
+      .map(mapper =>
+        mapper?.(source).map(item =>
+          path.extname(item)
+            ? item
+            : ['ts', 'tsx', '.d.ts', 'js'].map(ext => `${item}.${ext}`),
+        ),
+      )
+      .flat(2)
+      .filter(isFile)
   }
 
   if (retry && paths.length === 0) {

